eocker/src/lib.rs

```diff
@@ -1,10 +1,49 @@
+use crypto::{digest::Digest, sha2::Sha256};
+use flate2::{write::GzEncoder, Compression};
 use serde::{Deserialize, Serialize};
-use std::{collections::HashMap, time};
+use std::{collections::HashMap, error::Error, time};
 use types::MediaType;
 
 pub mod digest;
 pub mod types;
 
+#[derive(Debug)]
+pub struct Layer {
+    pub content: Vec<u8>,
+    pub descriptor: Descriptor,
+}
+
+impl Layer {
+    pub fn new_for_content(content: &[u8], path: &str) -> Result<Layer, Box<dyn Error>> {
+        let mut header = tar::Header::new_gnu();
+        header.set_path(path).unwrap();
+        header.set_size(content.len() as u64);
+        header.set_mode(0o644);
+        header.set_cksum();
+        let enc = GzEncoder::new(Vec::new(), Compression::fast());
```

Comment:
One thing that we will need here for even the simplest example is getting diff ID for uncompressed archive.

Reply:
Refactored this a bit so that we build uncompressed archive first, then calculate diff ID, then compress and get calculate digest, rather than passing the gz encoder as write source to tar builder.